tests/specs/lemmas.k

```diff
@@ -30,6 +30,8 @@ module LEMMAS
     rule _BUF [ _ .. W ] => .ByteArray requires W ==Int 0                    [simplification]
     rule  BUF [ 0 .. W ] => BUF        requires W ==Int #sizeByteArray(BUF)  [simplification]
 
+    rule (#buf(DATA_LEN, DATA) ++ _BUF) [0 .. DATA_LEN] => #buf( DATA_LEN , DATA ) [simplification]
+
```

Comment:
Actually, I think the three rules below this rule should subsume this one completely. Specifically the second rule should, starting on line 39.

So I'm actually surprised it doesn't work. I think we should figure out why the below lemmas are not applying. It could be that it cannot show `#range(0 <= START < #sizeByteArray(#buf(DATA_LEN, DATA)))`, or that it cannot show `START +Int WIDTH <=Int #sizeByteArray(DATA_LEN, DATA)`. You should:

- Remove this added lemma.
- Add functional claims that the two side-conditions above simplify to `true`, in addition to the one you already added above.
- Add lemmas which make those two side-conditions simplify to `true`.

Then the original claim should also simplify out, but without duplicating lemmas for `(_ ++ _) [_ .. _]`.

Reply:
Replacing the lemma
```
rule (BUF1 ++_BUF2)[START .. WIDTH] => (BUF1[START .. WIDTH])
      requires #range(0 <= START < #sizeByteArray(BUF1))
       andBool START +Int WIDTH <=Int #sizeByteArray(BUF1) [simplification]
```
with
```
rule (BUF1 ++ BUF2)[START .. WIDTH] => (BUF1[START .. #sizeByteArray(BUF1) -Int START]) ++ (BUF2[0 .. START +Int WIDTH -Int #sizeByteArray(BUF1)])
      requires 0 <=Int START 
        andBool 0 <=Int WIDTH
        andBool START +Int WIDTH <=Int #sizeByteArray(BUF1)
        [simplification]
```
seems to be a viable solution, since it's requirements are weaker and it makes the following claim pass:
```
claim <k> runLemma  ( (#buf (SIZE, DATA) ++ _BUF) [ 0 .. SIZE ] )
           => doneLemma ( #buf (SIZE, DATA) ) </k>
      requires #range(0 <= DATA < 2 ^Int (SIZE *Int 8))
       andBool 0 <=Int SIZE
```